labs/lab-2-javascript-game/script.js

```diff
@@ -0,0 +1,171 @@
+  // Global variables. Change these to alter the game.
+  const initUserHP = 40;
+  const initEnemyHP = 10;
+  const winsNeeded = 5;
+  const healCount = 2;
+
+  class Character {
+    constructor(name, health) {
+      this.name = name;
+      this.health = health;
+      this.maxHealth = health;
+      this.healsRemaining = healCount;
+      this.winCount = 0;
+    }
+
+    generateAttackDamage() {
+      const min = 1;
+      const max = 3;
+      return (Math.floor(Math.random() * (max - min + 1) + min));
+    }
+
+    heal() {
+      const min = 1;
+      const max = 10;
+      this.health = this.health + (Math.floor(Math.random() * (max - min + 1) + min));
+      this.health = Math.min(this.health, this.maxHealth);
+      this.healsRemaining = this.healsRemaining - 1;
+      var healPercent = (100 * this.healsRemaining / healCount).toString() + "%"
+      document.getElementById("heals").style.width = healPercent;
+      document.getElementById("heals-text").innerText = this.healsRemaining + " / " + healCount;
+    }
+
+    wins(enemy, quit, disableButtons) {
+      this.winCount = this.winCount + 1;
+      var winPercent = (100 * this.winCount / winsNeeded).toString() + "%"
+      document.getElementById("wins").style.width = winPercent;
+      document.getElementById("wins-text").innerText = this.winCount + " / " + winsNeeded;
+      if (this.health === 0) {
+        this.loses(enemy, quit, disableButtons);
+      } else if (this.winCount !== winsNeeded) {
+        document.getElementById("battle-message").innerText = this.name + " beat " + enemy.name + "! You need to win " + (winsNeeded - this.winCount) + " more round(s).";
+        enemy.health = initEnemyHP;
+      } else {
+        document.getElementById("battle-message").innerText = this.name + " has defeated " + enemy.name + "! Peace has been restored upon the land. \n(The game will restart automatically in 10 seconds.)";
+        setTimeout(quit, 10000);
+        disableButtons();
+      }
+    }
+
+    loses(enemy, quit, disableButtons) {
+      if (this.health === 0 && this.winCount === winsNeeded) {
+        document.getElementById("battle-message").innerText = "The battle ended in a draw! (The game will restart automatically in 10 seconds.)";
+        setTimeout(quit, 10000);
+        disableButtons();
+      } else {
+        document.getElementById("battle-message").innerText = this.name + " has been defeated by " + enemy.name + "! All hope is lost. \n(The game will restart automatically in 10 seconds.)";
+        setTimeout(quit, 10000);
+        disableButtons();
+      }
+    }
+
+    percentHealth() {
+      return (100 * this.health / this.maxHealth);
+    }
+
+    percentHealthStr() { // Needed for health-bar width
+      return this.percentHealth().toString() + "%";
+    }
+
+    percentHeals() {
+      return (100 * this.healsRemaining / healCount);
+    }
+
+    percentWinStr() {
+      return this.percentHeals().toString() + "%";
+    }
+
+    setHealthColor() {
```

Comment:
`set` here is not so hot. The fn returns the color, it doesn't set it -- `healthColor` would be better.